### PR TITLE
Request for code review

### PR DESCRIPTION
**Pull request by anisak000**
This repository has code based on that from the textbook chapter on Classes. I wrote a test for the Quarks code.

### CODE REVIEW

File: Particles.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# import the particle module
+
+# import particle as p
+# print(p.Particle.roar)
+
+import particle as p # another way of doing the same thing
+higgs = p.Particle()
+print(higgs.roar)
```
Comment (ECGonzales) [inline Particles.py]: Add blank line to end of file

**Review by ECGonzales — COMMENTED** [review]
Remove.pyc and .cache files from git

File: Instances.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# import the Particle class from the particle module
+import particle as p
+
+# create an empty list to hold observed particle data
+obs = []
+
+# append the first particle; by the way, append means to add
+obs.append(p.Particle())
```
Comment (dcuffari) [inline Instances.py]: running this code fails due to the missing argument in Particle(). This method is currently defined in particle.py as requiring 4 arguments; self, charge, mass, position. So the line should look like, 

obs.append(p.Particle(1,1,1))

Making this change allowed for the code to run correctly.

File: Instances.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+# assign its position
+obs[0].r = {'x': 100.0, 'y': 38.0, 'z': -42.0}
+
+# append the second particle
+obs.append(p.Particle())
```
Comment (dcuffari) [inline Instances.py]: This also requires 4 arguments (counting self).

File: Quarks/test_quark.py [hunk 4]
```diff
@@ -4,20 +4,19 @@
 t = Quark()
 
 def test_quark_1():
-    assert if self.flavor == "up":
-            self.flavor = "down"
+    assert t.flip("up") == "down"
```
Comment (dcuffari) [inline Quarks/test_quark.py]: flip only takes one argument (self).

Comment (dcuffari) [inline Quarks/test_quark.py]: A way that works is the following,

t.flavor = "up"
t.flip()
assert t.flavor == "down"

This passes and tests the method.

**Review by dcuffari — CHANGES_REQUESTED** [review]

**Comment by anisak000** [comment]
I have implemented all required changes. The tests work too. Thanks for the generous code review!

**Pull request merged**